plugin/trace/trace.go

```diff
@@ -0,0 +1,205 @@
+}
+
+func newTraceHandler(next http.Handler, t *Trace) (*oxytrace.Tracer, error) {
+	writer, err := newWriter(t.Addr)
+	if err != nil {
+		return nil, err
+	}
+	return oxytrace.New(next, writer, oxytrace.RequestHeaders(t.ReqHeaders...), oxytrace.ResponseHeaders(t.RespHeaders...))
+}
+
+// FromOther creates and validates Trace plugin instance from serialized format
+func FromOther(t Trace) (plugin.Middleware, error) {
+	return New(t.Addr, t.ReqHeaders, t.RespHeaders)
+}
+
+// FromCli creates a Trace plugin object from command line
+func FromCli(c *cli.Context) (plugin.Middleware, error) {
+	return New(c.String("addr"), c.StringSlice("reqHeader"), c.StringSlice("respHeader"))
+}
+
+// GetSpec returns all information neccessary for Vulcand to plugin this extension
+func GetSpec() *plugin.MiddlewareSpec {
+	return &plugin.MiddlewareSpec{
+		Type:      Type,
+		FromOther: FromOther,
+		FromCli:   FromCli,
+		CliFlags:  CliFlags(),
+	}
+}
+
+// CliFlags is used to add command-line arguments to the CLI tool - vctl
+func CliFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			Name:  "addr",
+			Usage: "Address of the output, e.g. syslog:///tmp/out.sock",
+		},
+		cli.StringSliceFlag{
+			Name:  "reqHeader",
+			Usage: "if provided, captures headers from requests",
+			Value: &cli.StringSlice{},
+		},
+		cli.StringSliceFlag{
+			Name:  "respHeader",
+			Usage: "if provided, captures headers from response",
+			Value: &cli.StringSlice{},
+		},
+	}
+}
+
+func newWriter(addr string) (io.Writer, error) {
+	u, err := url.Parse(addr)
+	if u.Scheme != "syslog" {
+		return nil, fmt.Errorf("unsupported scheme '%v' currently supported only 'syslog'", u.Scheme)
+	}
+	pr, err := parseSyslogPriority(u)
+	if err != nil {
+		return nil, err
+	}
+	if u.Host != "" {
+		return syslog.Dial("udp", u.Host, pr, parsePrefix(u))
+	} else if u.Path != "" {
+		return syslog.Dial("unixgram", u.Path, pr, parsePrefix(u))
+	} else if u.Host == "" && u.Path == "" {
+		return syslog.Dial("", "", pr, parsePrefix(u))
+	}
+	return nil, fmt.Errorf("unsupported address format: %v", addr)
+}
+
+func parsePrefix(u *url.URL) string {
+	t := u.Query().Get("prefix")
+	if t != "" {
+		return t
+	}
+	return SyslogPrefix
+}
+
+func parseSyslogPriority(u *url.URL) (syslog.Priority, error) {
+	vals := u.Query()
+	pr, err := sevToString(vals.Get("sev"))
+	if err != nil {
+		return 0, err
+	}
+	f, err := fToString(vals.Get("f"))
+	if err != nil {
+		return 0, err
+	}
+	return pr | f, nil
+}
+
+func sevToString(sev string) (pr syslog.Priority, err error) {
+	switch sev {
+	case "ALERT":
+		pr |= syslog.LOG_ALERT
+	case "CRIT":
+		pr |= syslog.LOG_CRIT
+	case "ERR":
+		pr |= syslog.LOG_ERR
+	case "WARNING":
+		pr |= syslog.LOG_WARNING
+	case "NOTICE":
+		pr |= syslog.LOG_NOTICE
+	case "INFO":
+		pr |= syslog.LOG_INFO
+	case "DEBUG", "":
+		pr |= syslog.LOG_DEBUG
+	default:
+		return pr, fmt.Errorf("uknown severity: %v", sev)
```

Comment:
You return `pr` in case of error here, but `0`, not `f`, in the function below (line 200) for seemingly the same situation. Not sure this matters, but just wanted to point it out in case you want to make this consistent.